bulk_indexer.go

```diff
@@ -257,52 +271,82 @@ func (b *bulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 		return resp, fmt.Errorf("error decoding bulk response: %w", err)
 	}
 
-	buf := make([]byte, 1024)
+	if b.maxDocumentRetry != 0 {
+		buf := make([]byte, 1024)
+
+		for k := range b.retryCounts {
+			found := false
+			for _, res := range resp.FailedDocs {
+				if res.Position == k {
+					found = true
```

Comment:
Do we need to keep iterating if the position matches?
```suggestion
					found = true
					break
```